src/rust/ide/view/src/searcher/icons.rs

```diff
@@ -63,6 +63,16 @@ fn grid(stroke_width:f32,cell_size:f32) -> AnyShape {
     (horizontal + vertical).repeat((cell_size.px(),cell_size.px())).into()
 }
 
+/// A path following the given points. All joints and endpoints are rounded.
+fn path(width:f32, points:&[(f32,f32)]) -> AnyShape {
+    let points     = points.iter().map(|(x,y)| (x.px(),y.px()));
+    let mut result : AnyShape = EmptyShape().into();
+    for (start,end) in points.clone().zip(points.skip(1)) {
+        result = (result + Segment(start,end,width.px())).into();
+    }
+    result
+}
```

Comment:
That is so cool implementation!

Reply:
It should be refactored somewhere where we define shapes in general. It is very useful!

Reply:
@wdanilo I am not sure where the definition should go. Should it be inside `display::shape::compound` or `display::shape::primitive`?

Reply:
lets move it to `display::shape::compound::path` module (file named path.rs).

Reply:
Done